src/test/java/qna/domain/AnswerRepositoryTest.java

```diff
@@ -1,22 +1,45 @@
 package qna.domain;
 
+import static qna.domain.UserTest.*;
 import static org.assertj.core.api.Assertions.assertThat;
+import static qna.domain.QuestionTest.Q1;
 
 import java.util.List;
 import java.util.Optional;
 
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
 
 @DataJpaTest
-public class AnswerTest {
-    public static final Answer A1 = new Answer(UserTest.JAVAJIGI, QuestionTest.Q1, "Answers Contents1");
-    public static final Answer A2 = new Answer(UserTest.SANJIGI, QuestionTest.Q1, "Answers Contents2");
+public class AnswerRepositoryTest {
+    public static final Answer A1 = new Answer(JAVAJIGI, Q1, "Answers Contents1");
+    public static final Answer A2 = new Answer(SANJIGI, Q1, "Answers Contents2");
 
     @Autowired
-    AnswerRepository answerRepository;
+    private AnswerRepository answerRepository;
+
+    @Autowired
+    private QuestionRepository questionRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @BeforeEach
+    void setup() {
+        userRepository.save(JAVAJIGI);
+        questionRepository.save(Q1);
+    }
```

Comment:
```suggestion
    @BeforeEach
    void setup() {
        User user = userRepository.save(JAVAJIGI);
        List<User> javajigis = userRepository.findAll();
        questionRepository.save(Q1);
    }
```

@AfterEach를 제거한 뒤에 위처럼 저장한 user 데이터를 확인해보면  
user 테이블에 데이터는 계속 지워지지만 id 가 계속 증가하는 것을 확인하실 수 있습니다 

DataJpaTest 는 테스트가 끝나면 rollback을 진행해주는데, 
autoIncrement 가 증가한 값까지는 rollback 해주지 않기 때문입니다 

Q1, A1 등 테스트를 위한 값을 선언하실 때, static 으로 선언해둔 JAVAJIGI 를 WRITER로 매핑하셨는데요 
이 JAVAJIGI 는 1L을 아이디로 갖는데 2번째 테스트 이후로는 
1L을 아이디로 갖는 유저는 존재하지 않기 때문에 에러가 난 것으로 보입니다 

static으로 선언한 값을 재사용하면 작성 시점에는 편할 수 있지만 그 값의 변경사항이 다른 테스트에도 영향을 줄 수 있습니다 
테스트는 다른 변경사항과는 독립적으로 성공을 보장해야하므로 다른 테스트에서 사용한 값을 참조하지 않도록 해보시죠 !

Reply:
맞습니다!! 수정 후 테스트 케이스가 잘 동작 하는 것을 확인하였습니다! 감사합니다!